src/net/sourceforge/kolmafia/session/GreyYouManager.java

```diff
@@ -0,0 +1,736 @@
+package net.sourceforge.kolmafia.session;
+
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.TreeMap;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import net.sourceforge.kolmafia.KoLCharacter;
+import net.sourceforge.kolmafia.KoLConstants;
+import net.sourceforge.kolmafia.Modifiers;
+import net.sourceforge.kolmafia.MonsterData;
+import net.sourceforge.kolmafia.objectpool.SkillPool;
+import net.sourceforge.kolmafia.persistence.AdventureDatabase;
+import net.sourceforge.kolmafia.persistence.MonsterDatabase;
+import net.sourceforge.kolmafia.persistence.SkillDatabase;
+import net.sourceforge.kolmafia.request.CharSheetRequest;
+import net.sourceforge.kolmafia.request.UseSkillRequest;
+import net.sourceforge.kolmafia.utilities.StringUtilities;
+
+public class GreyYouManager {
+
+  // Every monster you absorb is listed on the Charsheet
+
+  public static final Set<MonsterData> absorbedMonsters = new HashSet<>();
+
+  public static void refreshAbsorptions() {
+    CharSheetRequest request = new CharSheetRequest();
+    request.run();
+  }
+
+  // Absorbed 5 adventures from a warwelf.<!-- 199 -->
+  private static final Pattern ABSORPTION_PATTERN =
+      Pattern.compile(
+          "Absorbed (.*?) from (?: an?| the |)(.*?)\\.<!-- ([\\d]+) -->", Pattern.DOTALL);
```

Comment:
seeing as you don't use the displayed name at all, I feel like you could merge the pronoun with the name (as in, just match it all under the same `(.*?)`).

This would avoid a potential issue where a monster has a strange pronoun. Also I think you have a potential double space issue with the pronouns as-is (it would only match "from \<space> \<space> the", for example).

Reply:
I fixed that. And I added tests to make sure I found it and added other error checks which revealed that I had a fair number of erroneous monster names. Tests help. ;)